src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -694,6 +701,13 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 elementTypes.Add(elementType);
             }
 
+            if (elementNameCount != 0 && elementNameCount != elementTypes.Count)
+            {
+                hasErrors = true;
+                // PROTOTYPE: need specific error for specifying some names but not all
+                Error(diagnostics, ErrorCode.ERR_AnonymousTypeDuplicatePropertyName, node);
```

Comment:
I think we need to have distinct errors for distinct error conditions, otherwise unit-tests might not be as reliable as they should be.